src/transform/src/union_cancel.rs

```diff
@@ -19,6 +19,18 @@ use mz_expr::MirRelationExpr;
 pub struct UnionBranchCancellation;
 
 impl crate::Transform for UnionBranchCancellation {
+    /// `UnionBranchCancellation` is recursion-safe, but this is not immediately trivial:
```

Comment:
nit: we will eliminate this method at the end of the enclosing epic, so if you want to preserve the comment for posterity I suggest to put it on top of the `pub struct UnionBranchCancellation` item. above or somewhere in the code where one would normally check for recursion safety.

Reply:
Thanks, done!